lib/seahorn/BvOpSem2.cc

```diff
@@ -1488,7 +1490,24 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemBase {
     }
 
     if (F.getName().equals("shadow.mem.global.init")) {
-      WARN << "Skipping initialization of a global: " << inst << "\n";
+      WARN << "initializing a global: " << inst << "\n";
+      Expr memOut = m_ctx.mkRegister(inst);
+      Expr memIn = m_ctx.getRegister(*CS.getArgument(1));
+      m_ctx.read(memIn);
+      setValue(inst, havoc(inst));
+
+      m_ctx.setMemReadRegister(memIn);
+      m_ctx.setMemWriteRegister(memOut);
+      m_ctx.setMemScalar(extractUniqueScalar(CS) != nullptr);
+
+      llvm::GlobalVariable *gv = (GlobalVariable *)((*CS.getArgument(2)).stripPointerCasts());
```

Comment:
to cast, use `dyn_cast`. See examples in code, and LLVM documentation on the web.
`dyn_cast` returns `nullptr` if the argument cannot be safely converted to requested type.